src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -277,18 +277,20 @@ private void UpdateReferenceLocationsTask(Task<IInlineRenameLocationSet> allRena
             AssertIsForeground();
 
             var asyncToken = _asyncListener.BeginAsyncOperation("UpdateReferencesTask");
-            _allRenameLocationsTask = allRenameLocationsTask;
-            allRenameLocationsTask.SafeContinueWithFromAsync(
+            _allRenameLocationsTask = allRenameLocationsTask.SafeContinueWithFromAsync(
                 async t =>
                 {
                     await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
```

Comment:
This is now going to cause deadlocks. _allRenameLocationsTask is used as an antecedent to _conflictResolutionTask; _conflictResolutionTask is waited on directly on the UI thread.

This is the standard Roslyn pattern where there are tasks that are pure computation (and thus safe to wait), and some are notification tasks. _allRenameLocationTask and _conflictResolutionTask are both of the form that they're computation tasks, and thus cannot have a jump like this.

Reply:
What data is QueueApplyReplacements using here?

Reply:
`QueueApplyReplacements` schedules `OpenTextBufferManager.ApplyConflictResolutionEdits`, which assumes `_referenceSpanToLinkedRenameSpanMap` is already updated. `_referenceSpanToLinkedRenameSpanMap` is updated in `RaiseSessionSpansUpdated`.

Reply:
The deadlocks should now by fixed by following JTF rules.